lib/internal/process/modules.js

```diff
@@ -1,17 +1,49 @@
 'use strict';
 
 const {
+  setImportModuleDynamicallyCallback,
   setInitializeImportMetaObjectCallback
 } = internalBinding('module_wrap');
 
+const { getURLFromFilePath } = require('internal/url');
+const Loader = require('internal/loader/Loader');
+const path = require('path');
+const { URL } = require('url');
+
+function normalizeReferrerURL(referrer) {
+  if (typeof referrer === 'string' && path.isAbsolute(referrer)) {
+    return getURLFromFilePath(referrer).href;
+  }
+  return new URL(referrer).href;
+}
+
 function initializeImportMetaObject(wrap, meta) {
   meta.url = wrap.url;
 }
 
-function setupModules() {
-  setInitializeImportMetaObjectCallback(initializeImportMetaObject);
-}
+exports.loaderPromise = new Promise((resolve, reject) => {
+  exports.setup = function() {
```

Comment:
This works (defining `setup` inside the promise) but has some side effects - for example synchronous throws from `setup` will now be converted to `loaderPromise` errors.

I'd do something like:

```js
let loaderPromiseResolve;
exports.loaderPromise = new Promise(resolve => {
  loaderPromiseResolve = resolve;
});
```

And then define `loaderPromise` as resolve, and use `loaderPromiseResolve` on the new `loaderPromise` (since you can pass a rejected promise to it and it'll reject the original promise).